Added a new Analysis navigation submenu

hotzenklotz opened the pull request:
As discussed in #8383, we have several navbar menu entries that should be available to all users that are currently hidden in the "Admin" section. I moved them into a separate sub menu called "Analysis" (I am open for alternative nameing suggestions)

<img width="1651" height="299" alt="Screenshot 2025-11-24 at 16 44 46" src="https://github.com/user-attachments/assets/0638aca8-6a0d-467f-9200-e2956678447d" />

<img width="1159" height="211" alt="Screenshot 2025-11-24 at 16 44 19" src="https://github.com/user-attachments/assets/c79da0f4-9ecf-46ad-83ee-dd41bf6489f0" />

Additionally:
- I updated the AI model list to have a small description and link to docs
- I updated the list views for Jobs and VX workflow to use the same basic layout as AI model view (flex box spacing) 
- I updated the AI model listing API to be available for all users

### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz

### Steps to test:
- Click through navbar


### Issues:
- fixes #8383
- fixes #5701

------
(Please delete unneeded items, merge only when none are left open)
- [x] Added changelog entry (create a `$PR_NUMBER.md` file in `unreleased_changes` or use `./tools/create-changelog-entry.py`)
- [ ] Added migration guide entry if applicable (edit the same file as for the changelog)
- [x] Updated [documentation](../blob/master/docs) if applicable
- [ ] Adapted [wk-libs python client](https://github.com/scalableminds/webknossos-libs/tree/master/webknossos/webknossos/client) if relevant API parts change
- [ ] Removed dev-only changes like prints and application.conf edits
- [x] Considered [common edge cases](../blob/master/.github/common_edge_cases.md)
- [ ] Needs datastore update after deployment

hotzenklotz commented:
@philippotto Ping

philippotto commented:
for the record, I encountered a problem on the models page during testing. see https://scm.slack.com/archives/C5AKLAV0B/p1764685103701089?thread_ts=1763999294.239859&cid=C5AKLAV0B

hotzenklotz commented:
@philippotto 
- I updated the AI model list to have a small description and link to docs. Also use `react-query` for API fetching 
- I updated the list views for Jobs and VX workflow to use the same basic layout as AI model view (flex box spacing) 
- I updated the AI model listing API to be available for all users
